realsense_camera/src/base_nodelet.cpp

```diff
@@ -182,15 +150,13 @@ namespace realsense_camera
     }
 
     // Print list of all cameras found
-    listCameras(num_of_cameras);
-
-    pnh_.getParam("serial_no", serial_no);
-    pnh_.getParam("usb_port_id", usb_port_id);
+    int type_count = listCameras(num_of_cameras);
```

Comment:
Done with a slight modification of using the camera_type_index.size() instead of type_count.